### PR TITLE
Move implementation to source files

### PR DESCRIPTION
**Pull request by wkaluza**

### CODE REVIEW

File: libs/ledger/include/ledger/upow/synergetic_execution_manager_interface.hpp [hunk 1]
```diff
@@ -46,36 +48,7 @@ class SynergeticExecutionManagerInterface
...
-    text = "Invalid block";
-    break;
-  case ExecStatus::CONTRACT_REGISTRATION_FAILED:
-    text = "Contract registration failed";
-    break;
-  }
-
-  return text;
-}
+char const *ToString(SynergeticExecutionManagerInterface::ExecStatus status);
```
Comment (denis-trapeznikov) [inline libs/ledger/include/ledger/upow/synergetic_execution_manager_interface.hpp:51]: Could be made `constexpr` if left here.

File: libs/oef-base/include/oef-base/utils/Uri.hpp [hunk 2]
```diff
@@ -59,12 +59,12 @@ class Uri
                    "\" port=", port, " path=\"", path, "\"");
   }
 
-  inline std::string GetSocketAddress() const
+  std::string GetSocketAddress() const
```
Comment (denis-trapeznikov) [inline libs/oef-base/include/oef-base/utils/Uri.hpp:62]: Here and below, bodies should be moved to cpps then.

File: libs/oef-core/src/comms/public_key_utils.cpp [hunk 3]
```diff
@@ -18,16 +18,14 @@
...
-
 RSAKey::RSAKey(const EvpPublicKey &evp_pk)
   : rsa_{nullptr}
 {
   ERR_load_crypto_strings();
   rsa_ = EVP_PKEY_get1_RSA(evp_pk.native_handle());
   if (rsa_ != nullptr)
   {
-    std::runtime_error(ERR_error_string(ERR_get_error(), NULL));
+    std::runtime_error(ERR_error_string(ERR_get_error(), nullptr));
```
Comment (denis-trapeznikov) [inline libs/oef-core/src/comms/public_key_utils.cpp]: Is it to be thrown? `runtime_error` is a class.

**Review by denis-trapeznikov — CHANGES_REQUESTED** [review]

File: libs/oef-base/include/oef-base/utils/Uri.hpp [hunk 4]
```diff
@@ -59,12 +59,12 @@ class Uri
                    "\" port=", port, " path=\"", path, "\"");
   }
 
-  inline std::string GetSocketAddress() const
```
Comment (cyenyxe) [inline libs/oef-base/include/oef-base/utils/Uri.hpp:62]: Why not inlining these 2 methods?

**Review by cyenyxe — DISMISSED** [review]

File: libs/oef-base/include/oef-base/utils/Uri.hpp (hunk 4 above)
Comment (wkaluza) [reply libs/oef-base/include/oef-base/utils/Uri.hpp:62]: Member functions defined inside the class definition are implicitly `inline`.

File: libs/ledger/include/ledger/upow/synergetic_execution_manager_interface.hpp (hunk 1 above)
Comment (wkaluza) [reply libs/ledger/include/ledger/upow/synergetic_execution_manager_interface.hpp:51]: The performance benefit is likely to be negligible. Let's keep executable code out of headers if we can.

File: libs/oef-base/include/oef-base/utils/Uri.hpp (hunk 2 above)
Comment (wkaluza) [reply libs/oef-base/include/oef-base/utils/Uri.hpp:62]: The inline was redundant here. This is OEF code so I'd rather avoid merge conflict-prone changes like moving entire implementations to cpp files until after mainnet. The main focus of this PR is on ledger code.

File: libs/oef-core/src/comms/public_key_utils.cpp (hunk 3 above)
Comment (wkaluza) [reply libs/oef-core/src/comms/public_key_utils.cpp]: Huh, good spot. @qati @engineerkatie are we missing a `throw` here?

Comment (lrahmani) [reply libs/oef-core/src/comms/public_key_utils.cpp]: yes, that is correct
it should `throw` if `rsa_ == NULL`
btw, the file is months-old outdated.
updated files are on this branch https://github.com/fetchai/research-consensus/pull/5

Comment (wkaluza) [reply libs/oef-core/src/comms/public_key_utils.cpp]: Fixed @lrahmani

**Review by lrahmani — DISMISSED** [review]
I'll have to update the public_key_utils file in the near future

File: libs/ledger/src/upow/synergetic_execution_manager_interface.cpp [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+//------------------------------------------------------------------------------
+
+#include "ledger/upow/synergetic_execution_manager_interface.hpp"
+
+namespace fetch {
+namespace ledger {
+
+char const *ToString(SynergeticExecutionManagerInterface::ExecStatus status)
+{
+  using ExecStatus = SynergeticExecutionManagerInterface::ExecStatus;
```
Comment (denis-trapeznikov) [inline libs/ledger/src/upow/synergetic_execution_manager_interface.cpp:26]: `using SynergeticExecutionManagerInterface::ExecStatus;`

**Review by denis-trapeznikov — CHANGES_REQUESTED** [review]

File: libs/ledger/src/upow/synergetic_execution_manager_interface.cpp (hunk 5 above)
Comment (wkaluza) [reply libs/ledger/src/upow/synergetic_execution_manager_interface.cpp:26]: Done.

Comment (wkaluza) [reply libs/ledger/src/upow/synergetic_execution_manager_interface.cpp:26]: Reverted - ExecStatus is an enum.

Comment (denis-trapeznikov) [reply libs/ledger/src/upow/synergetic_execution_manager_interface.cpp:26]: Oh, sorry.

**Review by denis-trapeznikov — APPROVED** [review]

**Pull request merged**